okhttp-urlconnection/src/main/java/okhttp3/internal/huc/OkHttpURLConnection.java

```diff
@@ -587,6 +588,32 @@ private void setProtocols(String protocolsString, boolean append) {
     }
   }
 
+  static final class UnexpectedException extends IOException {
+    static final Interceptor INTERCEPTOR = new Interceptor() {
+      @Override public Response intercept(Chain chain) throws IOException {
+        try {
+          return chain.proceed(chain.request());
+        } catch (IOException e) {
+          throw e;
+        } catch (Error | RuntimeException e) {
+          throw new UnexpectedException(e);
+        }
+      }
+    };
+
+    public UnexpectedException(Throwable cause) {
+      super(cause);
+    }
+  }
+
+  /** Throws {@code throwable} as either an IOException, RuntimeException, or Error. */
+  private static IOException propagate(Throwable throwable) throws IOException {
```

Comment:
The unused return type exists to make the callsites happy . . . 
https://publicobject.com/2008/05/calling-method-that-always-throws.html